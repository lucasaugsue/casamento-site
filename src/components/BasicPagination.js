import React, { forwardRef } from 'react';
import Pagination from '@mui/material/Pagination';

const BasicPagination = forwardRef(({ page, count, handleChange }, ref) => (
    <Pagination
        page={page}
        color="primary"
        count={count || 10}
        onChange={handleChange}
        ref={ref}
    />
));

export default BasicPagination;
