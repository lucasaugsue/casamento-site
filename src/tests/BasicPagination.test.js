import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import BasicPagination from '../components/BasicPagination';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';

jest.mock('@mui/material/Pagination', () => {
    return (props) => (
      	<div
			data-testid="pagination-text"
			onClick={props.onChange}
		>
			{props.page}
			{props.count}
      	</div>
    );
});

describe('<BasicPagination/>', () => {
    describe('Renderização', () => {
        test('deve renderizar BasicPagination corretamente', async () => {
			const handleChange = jest.fn();
			const tree = renderer.create(
				<BasicPagination page={1} count={10} handleChange={handleChange} />
			).toJSON();

			// Verifica se o snapshot corresponde ao esperado
			expect(tree).toMatchSnapshot();
		});
	});
    
	describe('Interações', () => {
		test('deve chamar handleChange ao clicar na paginação', async () => {
			const handleChange = jest.fn();
	
			render(<BasicPagination page={1} count={10} handleChange={handleChange} />);
	
			fireEvent.click(screen.getByTestId('pagination-text'));
			await waitFor(() => {
				expect(handleChange).toHaveBeenCalled();
			});
		});
	});
});
