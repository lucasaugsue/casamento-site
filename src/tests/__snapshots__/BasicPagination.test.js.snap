// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderiza BasicPagination corretamente 1`] = `
<div
  onClick={[MockFunction]}
>
  1
  10
</div>
`;
